lib/puppet/util/windows/registry.rb

```diff
@@ -232,9 +232,9 @@ def read(key, name_ptr, *rtype)
         begin
           case type
             when Win32::Registry::REG_SZ, Win32::Registry::REG_EXPAND_SZ
-              result = [ type, data_ptr.read_wide_string(string_length) ]
+              result = [ type, sanitize(data_ptr.read_wide_string(string_length)) ]
             when Win32::Registry::REG_MULTI_SZ
-              result = [ type, data_ptr.read_wide_string(string_length).split(/\0/) ]
+              result = [ type, data_ptr.read_wide_string(string_length).split(/\0/).map!(&:scrub!) ]
```

Comment:
Per comment from Josh, `read_wide_string` already produces valid UTF-8, so no need to scrub.